frontend/packages/dev-console/src/components/add/layout/MasonryLayout.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import Measure from 'react-measure';
+import { Masonry } from './Masonry';
+import './MasonryLayout.scss';
+
+type MasonryLayoutProps = {
+  columnCount?: number;
+  children: React.ReactElement[];
+};
+
+const DEFAULT_COLUMN_WIDTH = 300;
+
+export const MasonryLayout: React.FC<MasonryLayoutProps> = ({ children, columnCount }) => {
+  const [width, setWidth] = React.useState<number>(0);
+  const defaultColumnCount = React.useMemo(
+    () => (width ? Math.floor(width / DEFAULT_COLUMN_WIDTH) || 1 : null),
+    [width],
+  );
+  const columns = columnCount || defaultColumnCount;
+
+  return (
+    <Measure bounds onResize={(contentRect) => setWidth(contentRect.bounds?.width)}>
+      {({ measureRef }) => (
+        <div className="odc-masonry-container" ref={measureRef}>
+          {columns ? <Masonry columnsCount={columns}>{children}</Masonry> : null}
+        </div>
+      )}
+    </Measure>
+  );
+};
```

Comment:
Just found this while debugging the MasonryLayout issue. `columnCount` is not used by the AddCardSection, so you can remove this and optional replace this with an width prop.

```suggestion
type MasonryLayoutProps = {
  columnWidth: number; // not optional, so you need to change AddCardSection as well
  children: React.ReactElement[];
};

export const MasonryLayout: React.FC<MasonryLayoutProps> = ({ columnWidth, children }) => {
  const [width, setWidth] = React.useState<number>(0);
  const columnCount = React.useMemo(() => (width ? Math.floor(width / columnWidth) || 1 : null), [
    columnWidth,
    width,
  ]);

  return (
    <Measure bounds onResize={(contentRect) => setWidth(contentRect.bounds?.width)}>
      {({ measureRef }) => (
        <div className="odc-masonry-container" ref={measureRef}>
          {columnCount ? <Masonry columnsCount={columnCount}>{children}</Masonry> : null}
        </div>
      )}
    </Measure>
  );
};
```